File: ssm-sdk-client/src/test/java/org/civis/blockchain/ssm/client/SsmClientTestBuilder.java
package org.civis.blockchain.ssm.client;

import java.io.IOException;

public class SsmClientTestBuilder {

    public static SsmClient build() throws IOException {
        String configFileName = getConfigFileName();
        return SsmClient.fromConfigFile(configFileName);
    }

    private static String getConfigFileName() {
        String profile = System.getenv("PROFILES_ACTIVE");
        System.out.println("////////////////////////////");
        System.out.println(profile);
        System.out.println("////////////////////////////");
        if(profile.equalsIgnoreCase("gitlab")) {
            return "ssm-client-gitlab.properties";
        }
        return "ssm-client.properties";
    }
}

File: ssm-sdk/ssm-sdk-core/src/test/kotlin/ssm/sdk/client/SsmClientTestBuilder.kt
package ssm.sdk.client

import java.io.IOException
import ssm.sdk.core.SsmServiceFactory

object SsmClientTestBuilder {

//	private val uuid = UUID.randomUUID().toString()
//	private const val NETWORK = "bclan-it/"

	@Throws(IOException::class)
	fun build(): SsmServiceFactory {
		return SsmServiceFactory.builder(configFileName).withChannelId("sandbox").withSsmId("ssm")
	}

	private val configFileName: String
		get() {
			val profile = System.getenv("PROFILES_ACTIVE")
			println("////////////////////////////")
			println(profile)
			println("////////////////////////////")
			return if ("gitlab".equals(profile, ignoreCase = true)) {
				"ssm-client-gitlab.properties"
			} else "ssm-client.properties"
		}
}
